lib/core/scheduler.rb

```diff
@@ -29,7 +30,11 @@ def when_triggered
       loop do
         connector_settings.each do |cs|
           if sync_triggered?(cs)
-            yield cs, :sync
+            if ElasticConnectorActions.find_current_job(cs.id).nil?
+              yield cs, :sync
+            else
+              Utility::Logger.debug("Sync job for #{cs.formatted} is already running, skipping...")
```

Comment:
would it skip this whole window, or just skip this loop?
Like, let's say jobs should sync every 60 minutes, and a sync takes 61 minutes to finish. Does the next sync start at 62 minutes, or at 120 minutes or at 121 minutes? And does the python framework have the same behavior?

Reply:
As to the python one, it's difficult to say. I am not great with py, and as I was looking at the codebase, I couldn't locate any code that would check for the job existence. 🤦 Perhaps I'm missing something because @tarekziade definitely mentioned that this check was there...

Reply:
👍 I agree that this is probably the best option